google/cloud/forseti/services/scanner/dao.py

```diff
@@ -377,4 +408,30 @@ def initialize(engine):
     Args:
         engine (object): Database engine to operate on.
     """
+    # Create tables if not exists.
     BASE.metadata.create_all(engine)
+    BASE.metadata.bind = engine
+
+    # Update schema changes.
+    # Find all the child classes inherited from declarative base class.
+    base_subclasses = _find_subclasses(BASE)
+
+    # Find all the Table objects for each of the classes.
+    # The format of tables is: {table_name: Table object}.
+    tables = BASE.metadata.tables
+
+    schema_update_method_name = 'schema_update'
+
+    for subclass in base_subclasses:
+        schema_update = getattr(subclass, schema_update_method_name, None)
+        if callable(schema_update) and subclass.__tablename__ in tables:
+            LOGGER.info('Updating table %s', subclass.__tablename__)
+            # schema_update will require the Table object.
+            schema_update(tables.get(subclass.__tablename__))
```

Comment:
As discussed, it would be awesome if the running of the schema update can be moved outside of the forseti application, so that the db update operation doesn't have to run everytime the forseti application starts up, and users can have the chance to backup their database before their db schema is actually updated.  This will also require a documentation change to our upgrade instruction as well.